packages/grid/_modules_/grid/hooks/features/columns/useColumns.ts

```diff
@@ -21,23 +21,27 @@ function mapColumns(
   columns: Columns,
   columnTypes: ColumnTypesRecord,
   viewportWidth: number,
+  withCheckboxSelection = false,
 ): Columns {
+  let availableViewportWidth = withCheckboxSelection
+    ? viewportWidth - checkboxSelectionColDef.width!
```

Comment:
what if the user resizes the checkbox column?

Reply:
Not a problem, resizing it behaves just like resizing any other column 😉